### DISCUSSION TITLE
Running pretrained models on batches of data

### DISCUSSION THREAD

**Discussion by jonrein** [pyannote/pyannote-audio#663] [Q&A]
The tutorials give some really nice examples on how to apply a pretrained model to a single test file, but it's not clear to me how to apply a pretrained model to a large set of my own data (parallelized in batches to speed up processing).  One of the tutorials (https://github.com/pyannote/pyannote-audio/tree/master/tutorials/pretrained/pipeline) mentions defining one's own protocol and points to the data prep tutorial (https://github.com/pyannote/pyannote-audio/tree/master/tutorials/data_preparation), but that tutorial seems to be mostly related to training.  What is required to create a protocol for unannotated test data?

Thanks!

**Comment by hadware** [comment]
You should define your own [pyannote database protocol](https://github.com/pyannote/pyannote-database) (as you'll see in the README, there is a couple of ways to do this), and leave the train and dev sets empty (which is fine for pyannote). If i'm not mistaken, pyannote doesn't care if audio files in the test set don't contain annotations (although, you obviously won't be able to score the performance of the model).
